src/options.jl

```diff
@@ -425,8 +425,21 @@ function check_and_add_approximation!(dict::Dict{Symbol,Any},
     check_aliases!(dict, dict_copy, [:ε])
     check_aliases!(dict, dict_copy, [:set_type])
     ε = haskey(dict, :ε) ? dict[:ε] : Inf
-    set_type = haskey(dict, :set_type) ? dict[:set_type] :
-        (ε < Inf ? HPolygon : Hyperrectangle)
+
+    if haskey(dict, :set_type)
+        set_type = dict[:set_type]
+    else
+        if ε < Inf
+            set_type = HPolygon
+        else
+            if !haskey(dict, :partition)
```

Comment:
Merge `else` and `if`?

Reply:
could be merged, but i kept the branch `ε = Inf` below because it is conceptually different:

```julia
        if ε < Inf
            set_type = HPolygon
        else
            # do not support iterative refinement => ignore ε
            if !haskey(dict, :partition)
                set_type = Interval
                set_type_proj = Interval
            else
                set_type = Hyperrectangle
            end
        end
```